# PROYECTO 11 - ANALISIS BASADO EN EVENTOS

El proyecto se basará en el estudio de una empresa que vende productos alimenticios. El objetivo es investigar el comportamiento del usuario para la aplicación de la empresa.

### Paso 1. Abrir el archivo de datos y leer la información general

In [1]:
# Importación de librerías necesarias para trabajar:
import pandas as pd
import numpy as np 
import scipy as sp
from scipy import stats as st
import matplotlib.pyplot as plt
import seaborn as sns
from datetime import datetime
import datetime as dt
from scipy import stats
import plotly.express as px

In [7]:
#lectura e información del dataframe:

logs = pd.read_csv('logs_exp_us.csv', sep= '\s+', header=0,
                   dtype={'EventName':'category' , 'ExpId':'category'})

display(logs.head())
print("Cantidad de filas es: ", len(logs))
print("La estructura del dataset es: ", logs.shape)
print("Tipo de datos:")
logs.dtypes
print()
print("Información estadística del dataset:")
logs.info(memory_usage="deep")

EventName         DeviceIDHash  EventTimestamp ExpId
0         MainScreenAppear  4575588528974610257      1564029816   246
1         MainScreenAppear  7416695313311560658      1564053102   246
2  PaymentScreenSuccessful  3518123091307005509      1564054127   248
3         CartScreenAppear  3518123091307005509      1564054127   248
4  PaymentScreenSuccessful  6217807653094995999      1564055322   248

Cantidad de filas es:  244126
La estructura del dataset es:  (244126, 4)
Tipo de datos:

Información estadística del dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244126 entries, 0 to 244125
Data columns (total 4 columns):
 #   Column          Non-Null Count   Dtype   
---  ------          --------------   -----   
 0   EventName       244126 non-null  category
 1   DeviceIDHash    244126 non-null  int64   
 2   EventTimestamp  244126 non-null  int64   
 3   ExpId           244126 non-null  category
dtypes: category(2), int64(2)
memory usage: 4.2 MB


A primera instancia con este DF logs, obtenemos que esta compuesto por cuatro columnas con nombres que no están bien definidos para seguir o continuar una praxis correcta. Por otro lado, podemos apreciar que el conteo de los no nulos por la información del dataset sería de cero. Finalmente, cada entrada de registro es una acción de usuario o un evento.
+ EventName: nombre del evento.
+ DeviceIDHash: identificador de usuario unívoco.
+ EventTimestamp: hora del evento.
+ ExpId: número de experimento: 246 y 247 son los grupos de control, 248 es el grupo de prueba.

### Paso 2. Preparar los datos para el análisis

####  2.1 Cambia el nombre de las columnas y comprueba si hay tipos de datos y valores ausentes.